extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -251,6 +255,46 @@ export class ProjectsController {
 		}
 	}
 
+	/**
+	 * Deploys a project
+	 * @param treeNode a treeItem in a project's hierarchy, to be used to obtain a Project
+	 */
+	public async deployProject(context: Project | dataworkspace.WorkspaceTreeItem): Promise<void> {
+		const project: Project = this.getProjectFromContext(context);
+		try {
+			let deployProfile = await launchDeployDatabaseQuickpick(project);
+			if (deployProfile && deployProfile.deploySettings) {
+				let connectionUri: string | undefined;
+				if (deployProfile.localDbSetting) {
+					connectionUri = await this.deployService.deploy(deployProfile, project);
+					if (connectionUri) {
+						deployProfile.deploySettings.connectionUri = connectionUri;
+					}
+				}
+				if (deployProfile.deploySettings.connectionUri) {
+					const publishResult = await this.publishOrScriptProject(project, deployProfile.deploySettings, true);
+					if (publishResult && publishResult.success) {
+
+						// Update app settings if requested by user
+						//
+						await this.deployService.updateAppSettings(deployProfile);
+						if (deployProfile.localDbSetting) {
+							await this.deployService.getConnection(deployProfile.localDbSetting, true, deployProfile.localDbSetting.dbName);
+						}
+						vscode.window.showInformationMessage(constants.deployProjectSucceed);
+					} else {
+						vscode.window.showErrorMessage(constants.deployProjectFailed(publishResult?.errorMessage || ''));
+					}
+				} else {
+					vscode.window.showErrorMessage(constants.deployProjectFailed('Failed to open a connection to the deployed database'));
+				}
+			}
+		} catch (error) {
+			vscode.window.showErrorMessage(constants.deployProjectFailed(error));
```

Comment:
The constants expects a string - but typically we should strive for always having errors throw actual error objects instead of just straight strings. So if that isn't being done here then would suggest fixing that - and then regardless using utils.getErrorMessage to extract the correct error message (since we can't always depend on external libraries to follow this)